### DISCUSSION TITLE
repetition mask with leading fixed symbol

### DISCUSSION THREAD

**Discussion by daniil4udo** [uNmAnNeR/imaskjs#811] [Q&A] [answered]
## What problem you are trying to solve?

Create repetition mask with leading fixed symbol alike `/^\+\d+$/`.

As mentioned in the docs, **RegExp** mask wont work, but **Pattern** has fixed length

## Describe the solution you'd like

I think there is a possibility to implement it via Function mask, but my data is fetch from the server in the RegExp and processed in the client side to be able to be used as mask inputs.

## Describe alternatives you've considered

But maybe there is more sophisticated solution for fixed symbol with repetition mask?

**Comment by uNmAnNeR** [comment, marked as answer]
```
IMask(el, {
  mask: '+num',
  lazy: true,
  blocks: {
    num: {
      mask: /^\d*$/,
    }
  }
})
```

**Reply by daniil4udo** [reply]
Wonderful! Thank you @uNmAnNeR
